### PR TITLE
add comment lexing and tests, initial support of block and line comme…

### PR DESCRIPTION
**Pull request by pm-mck**
…nts, abandon all comments before parsing stage.

Adds comments to the lexer and simply skips them. Supports multiline/block and standard line comments.

Caveats:

1. Does not keep documentation comments, they are discarded along with the other comments.
2. File cannot end with a line comment - it must have a final newline character.

### CODE REVIEW

File: src/schema_lexer.xrl [hunk 1]
```diff
@@ -6,6 +6,8 @@ STRING          = [a-zA-Z0-9_\.]+
 BOOL            = (true|false)
 WS              = [\s\t]+
 NL              = [\n\r]+
+COMMENT         = //.[^\n]*\n
```
Comment (an2deg) [inline src/schema_lexer.xrl]: could be simplified to COMMENT = //[^\n]*

File: test/schemas/comment_test.fbs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// Line 1 comment
+
```
Comment (an0nym05) [inline test/schemas/comment_test.fbs:6]: Fails with this kind of comments:
```
//
// asdf
//
```

Error:
```
iex(5)> Eflatbuffers.Schema.parse!(File.read!("test/lilli-proto/ds_core/overrideevent.fbs"))
** (MatchError) no match of right hand side value: {:error, {3, :schema_lexer, {:illegal, ~c"//\n"}}, 3}
    (eflatbuffers 0.1.0) lib/eflatbuffers/schema.ex:41: Eflatbuffers.Schema.lexer/1
    (eflatbuffers 0.1.0) lib/eflatbuffers/schema.ex:28: Eflatbuffers.Schema.parse/1
    (eflatbuffers 0.1.0) lib/eflatbuffers/schema.ex:18: Eflatbuffers.Schema.parse!/1
    iex:5: (file)
```

**Review by an0nym05 — COMMENTED** [review]
Found so far one case that is not covered.

File: src/schema_lexer.xrl (hunk 1 above)
Comment (an0nym05) [reply src/schema_lexer.xrl]: @an2deg your recommended change doesn't fail with this style of comments:
```
//
// asdf
//
```

File: test/schemas/comment_test.fbs (hunk 2 above)
Comment (an0nym05) [reply test/schemas/comment_test.fbs:6]: added some immediate mixes and merging for now. all side-effects to be address while testing our set of contracts.

**Pull request merged**